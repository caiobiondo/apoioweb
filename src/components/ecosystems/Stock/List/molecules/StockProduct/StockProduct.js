import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  productImageStyles,
  StockItemProductImageWrapper,
  StockItemProductImageFallback,
  StockItemProductDetails,
  StockItemProductDetailsName,
  StockItemProductDetailsCode,
} from './StockProduct.styles';
import { Icon } from 'natura-ui';
import Img from 'react-image';
import ImageWithFallback from 'components/molecules/ImageWithFallback';

class StockProduct extends Component {
  state = {};

  renderItemProductImageFallback = () => {
    return (
      <StockItemProductImageFallback>
        <Icon file="ico_pictureless" />
      </StockItemProductImageFallback>
    );
  };

  render() {
    let productName, imageUrl, productCode;
    if (this.props.product) {
      productName = this.props.product.name;
      imageUrl = `http://rede.natura.net/image/sku/145x145/${this.props.product.productId}_1.jpg`;
      productCode = this.props.product.productId;
    } else {
      productName = 'Nenhum produto selecionado';
      imageUrl = null;
      productCode = null;
    }

    return (
      <StockItemProductImageWrapper>
        <ImageWithFallback
          fallbackImage="ico_pictureless"
          imageUrl={imageUrl}
          styles={productImageStyles}
        />
        <StockItemProductDetails>
          <StockItemProductDetailsName>{productName}</StockItemProductDetailsName>
          {productCode && (
            <StockItemProductDetailsCode>CÃ³digo: {productCode}</StockItemProductDetailsCode>
          )}
        </StockItemProductDetails>
      </StockItemProductImageWrapper>
    );
  }
}

StockProduct.propTypes = {
  product: PropTypes.object,
};

export default StockProduct;
