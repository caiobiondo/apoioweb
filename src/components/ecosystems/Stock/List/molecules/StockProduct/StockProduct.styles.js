import styled from 'styled-components';
import { gray200 } from 'styles/colors';
import { Responsive } from '@entria/components';
import { RobotoRegular } from 'styles/typography';

export const StockItemProductImageWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  margin-top: 30px;
  margin-left: 30px;
  margin-right: 30px;

  svg {
    margin: 0 35px 0 30px;
    width: 114px;
  }

  img {
    margin: 0;
    width: 145px;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin-top: 0;
    margin-left: 0;
    margin-right: 0;

    svg {
      margin: 0 15px 0 0;
      max-width: 80px;
      height: 75px;
    }

    img {
      margin: 0;
      max-width: 80px;
      height: 85px;
    }
  }
`;

export const StockItemProductImageFallback = styled.div`
  svg {
    fill: ${gray200};
    width: 114px;
  }

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    svg {
      width: 75px;
    }
  }
`;

export const StockItemProductDetails = styled.div`
  display: flex;
  flex-direction: column;
`;

export const StockItemProductDetailsName = styled.div`
  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 14px;
    line-height: 16px;
  }
`;

export const StockItemProductDetailsCode = styled.div`
  font-family: ${RobotoRegular};
  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    font-size: 12px;
  }
`;

export const StockItemProductLoadingWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 90px;

  @media (max-width: ${Responsive.VIEWPORT.MEDIUM}px) {
    margin-top: 0;
    height: 85px;
  }
`;

export const productImageStyles = {
  display: 'inline-block',
  width: '50%',
};
