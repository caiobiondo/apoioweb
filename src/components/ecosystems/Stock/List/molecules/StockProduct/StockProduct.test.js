import React from 'react';
import StockProduct from './StockProduct';
import { shallow } from 'enzyme';

const setup = propOverrides => {
  const props = Object.assign(
    {
      product: {
        description: 'new product',
        name: 'new product',
        price: '10',
        productId: '1',
      },
    },
    propOverrides,
  );

  const result = shallow(<StockProduct {...props} />);

  return {
    props,
    result,
  };
};

describe('StockProduct', () => {
  it('renders correctly', () => {
    // given
    // when
    const { result } = setup();

    // then
    expect(result).toMatchSnapshot();
  });
});
