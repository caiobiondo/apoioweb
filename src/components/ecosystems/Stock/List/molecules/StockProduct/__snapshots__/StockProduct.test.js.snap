// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockProduct renders correctly 1`] = `
<StockProductstyles__StockItemProductImageWrapper>
  <Img
    decode={true}
    loader={
      <Loading
        background="#FFFFFF"
        visible={true}
        zIndex={9999}
      />
    }
    src="http://rede.natura.net/image/sku/145x145/1_1.jpg"
    unloader={
      <StockProductstyles__StockItemProductImageFallback>
        <Icon
          file="ico_pictureless"
        />
      </StockProductstyles__StockItemProductImageFallback>
    }
  />
  <StockProductstyles__StockItemProductDetails>
    <StockProductstyles__StockItemProductDetailsName>
      new product
    </StockProductstyles__StockItemProductDetailsName>
    <StockProductstyles__StockItemProductDetailsCode>
      <FormattedMessage
        id="stockProductCode"
        values={
          Object {
            "code": "1",
          }
        }
      />
    </StockProductstyles__StockItemProductDetailsCode>
  </StockProductstyles__StockItemProductDetails>
</StockProductstyles__StockItemProductImageWrapper>
`;
